src/tools/db-upgrade/DbUpgrader.cpp

```diff
@@ -82,10 +87,44 @@ Status UpgraderSpace::initSpace(const std::string& sId) {
   // Use readonly rocksdb
   readEngine_.reset(new nebula::kvstore::RocksEngine(
       spaceId_, spaceVidLen_, srcPath_, "", nullptr, nullptr, true));
-  writeEngine_.reset(new nebula::kvstore::RocksEngine(spaceId_, spaceVidLen_, dstPath_));
+  if (FLAGS_partIds.empty()) {
+    writeEngine_.reset(new nebula::kvstore::RocksEngine(spaceId_, spaceVidLen_, dstPath_));
+  }
 
   parts_.clear();
-  parts_ = readEngine_->allParts();
+
+  if (FLAGS_partIds.empty()) {
+    // find the specified partIds
+    parts_ = readEngine_->allParts();
+  } else {
+    std::vector<std::string> parts;
+    std::vector<PartitionID> partsAll;
+    folly::split(",", FLAGS_partIds, parts, true);
+    for (auto& part : parts) {
+      PartitionID partId;
+      try {
+        partId = folly::to<PartitionID>(part);
+      } catch (const std::exception& ex) {
+        LOG(ERROR) << "Cannot convert part id " << part;
+        return Status::Error("Cannot convert part id %s", part.c_str());
+      }
+      partsAll.emplace_back(partId);
+    }
+
+    // to check in rocksdb
+    for (auto part : partpartsAlls) {
```

Comment:
If the we can't find every part in `FLAGS_partIds` in local rocksdb , we should return error.

Reply:
Yeah, direct `LOG(FATAL)` and exit